src/buehlmann/buehlmann_model.rs

```diff
@@ -23,9 +29,28 @@ impl DecoModel for BuehlmannModel {
     fn step(&mut self, depth: &Depth, time: &Seconds, gas: &Gas) {
         let step = Step { depth, time, gas };
         self.depth = *step.depth;
+        self.gas = *step.gas;
```

Comment:
Is it required to pass first depth and gas to step object to later on assign it to self?